skbio/diversity/_driver.py

```diff
@@ -180,6 +182,111 @@ def alpha_diversity(metric, counts, ids=None, validate=True, **kwargs):
     return pd.Series(results, index=ids)
 
 
+@deprecated(as_of='0.4.2-dev', until='0.5.0',
+            reason=('The return type is unstable. Developer caution is '
+                    'advised. The resulting DistanceMatrix object may include '
+                    'zeros when distance has not been calculated, and '
+                    'therefore can be misleading.'))
+def partial_beta_diversity(metric, counts, ids=None, validate=None,
+                           id_pairs=None, **kwargs):
+    """Compute distances only between specified ID pairs
+
+    Parameters
+    ----------
+    metric : callable
+        The pairwise distance function to apply. ``metric`` must be resolvable
+        by scikit-bio (e.g., UniFrac methods), or must be callable.
+    counts : 2D array_like of ints or floats
+        Matrix containing count/abundance data where each row contains counts
+        of OTUs in a given sample.
+    ids : iterable of strs
+        Identifiers for each sample in ``counts``.
+    id_pairs : iterable of tuple, optional
+        An iterable of tuples of IDs to compare (e.g., ``[('a', 'b'), ('a',
+        'c'), ...])``. If specified, the set of IDs described must be a subset
+        of ``ids``.
+    validate : ignored
+        This kwarg is preserved to maintain consistency with the beta_diversity
+        signature.
+    kwargs : kwargs, optional
+        Metric-specific parameters.
+
+    Returns
+    -------
+    skbio.DistanceMatrix
+        Distances between all pairs of samples (i.e., rows). The number of
+        rows and columns will be equal to the number of rows in ``counts``.
+
+        Warning: zeros in this matrix may represent a zero distance, or may
+        represent a pairwise distance that was not calculated.
+
+    Raises
+    ------
+    ValueError
+        If ``ids`` are not specified.
+        If ``id_pairs`` are not a subset of ``ids``.
+        If ``metric`` is not a callable.
+        If duplicates are observed in ``id_pairs``.
+
+    See Also
+    --------
+    skbio.beta_diversity
+    skbio.diversity.get_beta_diversity_metrics
+
+    """
```

Comment:
It's important to include b/c it converts `array_like` into an `ndarray`. I think we have an open issue somewhere to separate validation from input munging but for now `_validate_counts_matrix` has to be called. It's also doing important validations like making sure the length of `ids` matches `counts`. Since we're exposing `validate=True` as part of the API and referring users to `beta_diversity`'s docs, validation needs to match here.